### PR TITLE
WT-12471 Remove log-based incremental backup documentation

### PR DESCRIPTION
**Pull request by korteland**
This includes a few smaller fixes, two of which I'm not very sure about. So those bits could use a much more careful review.

I also took the liberty of fixing some of the grammar in the documentation adjacent to what I removed.

### CODE REVIEW

File: examples/c/ex_all.c [hunk 1]
```diff
@@ -1161,12 +1161,6 @@ backup(WT_SESSION *session)
     error_check(session->open_cursor(session, NULL, cursor, "target=(\"log:\")", &dup_cursor));
     /*! [backup log duplicate]*/
 
-    /*! [incremental backup]*/
-    /* Open the backup data source for log-based incremental backup. */
-    error_check(session->open_cursor(session, "backup:", NULL, "target=(\"log:\")", &cursor));
-    /*! [incremental backup]*/
-    error_check(cursor->close(cursor));
-
```
Comment (korteland) [inline examples/c/ex_all.c:1169]: This wasn't identified previously because we don't even call this function.

File: src/log/log_sys.c [hunk 2]
```diff
@@ -31,8 +31,7 @@ __wt_log_system_backup_id(WT_SESSION_IMPL *session)
      * If we're not logging or incremental backup isn't turned on or this version doesn't support
      * the system log record, we're done.
      */
-    if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED) ||
-      !FLD_ISSET(conn->log_flags, WT_CONN_LOG_INCR_BACKUP))
+    if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))
```
Comment (korteland) [inline src/log/log_sys.c:34]: This is a spot I wasn't sure about. I don't completely understand the original logic - the ticket that added it doesn't appear to be specific to log-based incremental backup, but it seems we need both of these flags to be set for us to generate the new log record.

Is it possible I misunderstood the `WT_CONN_LOG_INCR_BACKUP` flag and that this flag actually means "log that we did an incremental backup"?

**Review by ruby-oujo — APPROVED** [review]

File: src/cursor/cur_backup.c [hunk 3]
```diff
@@ -267,9 +267,6 @@ __curbackup_close(WT_CURSOR *cursor)
         conn->modified = true;
         WT_TRET(__wt_txn_checkpoint(session, cfg, true));
     }
-    /* Clear the flag on force stop after the completion of the checkpoint. */
-    if (F_ISSET(cb, WT_CURBACKUP_FORCE_STOP))
-        FLD_CLR(conn->log_flags, WT_CONN_LOG_INCR_BACKUP);
```
Comment (sueloverso) [inline src/cursor/cur_backup.c:272]: Although it is unfortunately named, this flag and this code need to stay. This code has to do with force stopping block-based incremental backup and whether a system log record needs to be written for the block-based backup IDs.

File: src/log/log_sys.c (hunk 2 above)
Comment (sueloverso) [reply src/log/log_sys.c:34]: Yes, per my earlier comment, this flag is whether to log the block-based IDs or not. This all needs to stay

**Review by sueloverso — COMMENTED** [review]
This needs to reduce what it changes and removes.

File: src/log/log_sys.c (hunk 2 above)
Comment (korteland) [reply src/log/log_sys.c]: Thanks for confirming - reinstated in d624f4f.

**Review by sueloverso — APPROVED** [review]
LGTM

**Pull request merged**